src/wrappers/themis/themispp/secure_session.hpp

```diff
@@ -63,26 +63,32 @@ namespace themispp{
     typedef std::vector<uint8_t> data_t; 
 
     secure_session_t():
-      _session(NULL){
+      _session(NULL),
+      _callback(NULL){
     }
 
     secure_session_t(const data_t& id, const data_t& priv_key, secure_session_callback_interface_t* callbacks):
       _session(NULL),
+      _callback(NULL),
       _res(0){
-      _callback.get_public_key_for_id=themispp::get_public_key_for_id_callback;
-      _callback.send_data=themispp::send_callback;
-      _callback.receive_data=themispp::receive_callback;
-      _callback.state_changed=NULL;
-      _callback.user_data=callbacks;
-      _session=secure_session_create(&id[0], id.size(), &priv_key[0], priv_key.size(), &_callback);
-      if(!_session)
-	throw themispp::exception_t("Secure Session failde creating");
+      _callback=new secure_session_user_callbacks_t();
+      _callback->get_public_key_for_id=themispp::get_public_key_for_id_callback;
+      _callback->send_data=themispp::send_callback;
+      _callback->receive_data=themispp::receive_callback;
+      _callback->state_changed=NULL;
+      _callback->user_data=callbacks;
+      _session=secure_session_create(&id[0], id.size(), &priv_key[0], priv_key.size(), _callback);
+      if(!_session){
+        delete _callback;
+        throw themispp::exception_t("Secure Session failed creating");
+      }
     }
 
     virtual ~secure_session_t(){
       if(_session){
         secure_session_destroy(_session);
```

Comment:
set nullptr to _session and will be nice (and in other places)

Reply:
I've added zeroing of pointer which are freed in destructor, just to be safe. There is no need to zero them in move assignment after freeing as they are immediately replaced with new values.